### ISSUE TITLE
wrong page break prohibition between text and block box

### ISSUE THREAD

**Issue by MurakamiShinyu** [vivliostyle/vivliostyle.js#1036]
Test sample: 
https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-bug1.html

```html
<style>
  b {
    display: block;
    border: 5px dashed blue;
    padding: 5px;
    margin: 15px;
  }
</style>
<body>
  <p>
    そして島と十字架とは、だんだん濃く黒くはっきりなって、ちらちらと燃えるように見えたのでした。
    あすこがほんとうの神さまがほんとうの神さまがほんとうの天上へさえ行ける切符だ。
    うすあかい河原なでしこの花が咲いている
    <b>
      汽車はもうだんだん早くなって、その前にしょんぼりひとりの子供が立って、
      森の中に立ち、黄金の円光をもったりんどうの花
    </b>
    いっぱいに舞いおりて来ましたけれども、どうしてもっと愉快になれないだろう。
  </p>
</body>
```

Test with Vivliostyle Viewer (v2.19.1):
- https://vivliostyle.github.io/viewer/v2.19.1/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-bug1.html

Result screenshot:
<img width="1001" alt="Screenshot 2022-11-10 at 18 20 33" src="https://user-images.githubusercontent.com/3324737/201051396-22b1aaa4-61a7-4d80-bf0d-30391dd45cf1.png">

In the test result, a page break occurs before the last line of the first text (`…の花が咲いている`) even though there is sufficient space on the bottom of the first page. It is as if page break is prohibited between the text (`…の花が咲いている`) and the block box (`<b>汽車はもう…</b>`, note: `display: block` is specified).

**Comment by MurakamiShinyu** [comment]
Display-math, i.e,. `<math display="block">…</math>` (in MathML) or `<span data-math-typeset="true">$$ … $$</span>` (in TeX Math), inside paragraph has same problem. Test samples:

- https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug1.html
  - Test with Vivliostyle Viewer (v2.19.1): https://vivliostyle.github.io/viewer/v2.19.1/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug1.html
- https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug2.html
  - Test with Vivliostyle Viewer (v2.19.1): https://vivliostyle.github.io/viewer/v2.19.1/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug2.html

**Comment by MurakamiShinyu** [comment]
Fixed in #1039.

Test with Vivliostyle Viewer (after this fix):
- https://vivliostyle-git-fix-issue1036-vivliostyle.vercel.app/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-bug1.html
- https://vivliostyle-git-fix-issue1036-vivliostyle.vercel.app/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug1.html
- https://vivliostyle-git-fix-issue1036-vivliostyle.vercel.app/#src=https://github.com/MurakamiShinyu/bugtest/blob/main/box-break/pbreak-block-math-bug2.html

**Comment by MurakamiShinyu** [comment]
I found this issue is duplicate of #611
